### PR TITLE
Users API: add the update endpoint

### LINKED ISSUE #1992: Add an update method to galaxy API, so the remote user name could be changed
In large galaxy instances changing user's email is a routine operation. It would be very useful to be able to change the remote user name in the galaxy via api and with even more convenience via bioblend methods. However, managers/users.py and by extension api/users.py doesn't provide an update method that bioblend or raw api requests could use. This request is being opened after a discussion with @nsoranzo.

### PR DESCRIPTION
**Pull request by carlfeberhard**
Implements update for the users API. Currently only username is deserialized but it should be easier to add more now.
- tests for both the deserializer and the API endpoint were added
- fixes some magic numbers and removes some TODOs

The user deserializer could be improved in the future by bypassing `validate_user_input` and removing the need to pass `trans` for the session.

Should close: #1992

### CODE REVIEW

File: test/unit/managers/test_UserManager.py [hunk 1]
```diff
@@ -12,9 +12,10 @@
 from six import string_types
 
 from galaxy import exceptions, model
+from galaxy.managers import base as base_manager
 from galaxy.managers import histories, users
 
-from .base import BaseTestCase
+from base import BaseTestCase
```
Comment (nsoranzo) [inline test/unit/managers/test_UserManager.py]: Please revert to include the dot (for Python3).

File: test/unit/managers/test_UserManager.py [hunk 2]
```diff
@@ -216,6 +217,45 @@ def test_anonymous( self ):
...
+class UserDeserializerTestCase( BaseTestCase ):
+
+    def set_up_managers( self ):
+        super( UserDeserializerTestCase, self ).set_up_managers()
+        self.deserializer = users.UserDeserializer( self.app )
+
+    def _assertRaises_and_return_raised( self, exception_class, fn, *args, **kwargs ):
+        try:
+            fn( *args, **kwargs )
+        except exception_class, exception:
```
Comment (nsoranzo) [inline test/unit/managers/test_UserManager.py]: `except exception_class as exception:`

**Comment by nsoranzo** [comment]
:+1:

File: lib/galaxy/managers/base.py [hunk 3]
```diff
@@ -736,7 +736,7 @@ class ModelDeserializer( HasAModelManager ):
     """
     # TODO:?? a larger question is: which should be first? Deserialize then validate - or - validate then deserialize?
 
-    def __init__( self, app, **kwargs ):
+    def __init__( self, app, validator=None, **kwargs ):
```
Comment (nsoranzo) [inline lib/galaxy/managers/base.py:739]: The new validator parameter does not seem to be used, is it for future functionalities?

Comment (carlfeberhard) [inline lib/galaxy/managers/base.py:739]: Future composition:
https://github.com/galaxyproject/galaxy/pull/2595/files#diff-d9bd7857711700c7d77c35956699096dR306
Up to this point ModelValidator was enough.

**Comment by dannon** [comment]
Looks good to me.

File: lib/galaxy/managers/base.py (hunk 3 above)
Comment (nsoranzo) [inline lib/galaxy/managers/base.py:739]: Thanks @carlfeberhard!

**Pull request merged**